Source/BugsnagConfiguration.m

```diff
@@ -106,6 +106,12 @@ -(instancetype)initWithApiKey:(NSString *)apiKey
     _breadcrumbs = [BugsnagBreadcrumbs new];
     _automaticallyCollectBreadcrumbs = YES;
     _autoTrackSessions = YES;
+    // Default to recording all error types
+    _enabledErrorTypes = BSGErrorTypesOOMs
```

Comment:
Enabling OOM detection by default should be guarded to only happen in release builds, to avoid triggering the heuristic when killing/restarting an app in Xcode or similar.